### DISCUSSION TITLE
Enforce a Whitelist of Values for Regex-based Path

### DISCUSSION THREAD

**Discussion by aress31** [remix-run/react-router#10745] [Q&A]
I would like to know if it is possible to enforce a list of allowed `:id` for a route without triggering the `errorComponent`, for any other values the `errorComponent` should be displayed. See below:

```js
{
  id: "project",
  path: "projects/:projectId",
  element: (
    <WithTabs
      tabs={["x", "y", "z"]}
    />
  ),
  children: createProjectRoutes(),
},
```

```js
const createProjectRoutes = () => {
  return [
    {
      path: ":viewId",
      element: <DataTableView />,
    },
  ];
};
```

In this case `:viewId` should only be allowed to take the following values:

- `x`
- `y`
- `z`

Is there a trivial way of achieving this behaviour?

**Comment by aress31** [comment]
Up!

**Comment by GabenGar** [comment]
How "trivial" do you need it be? Just validate and throw in a loader.

**Comment by aress31** [comment]
This is my workaround, but it's not very developer-friendly as it adds a lot of boilerplate code. For routes with dynamic paths, it would be nice to explicitly set a whitelist using react-router.

**Reply by GabenGar** [reply]
That's not a workaround, that's any procedure which depends on a user input. 
Short of inventing a schema DSL, there is no more trivial way beyond writing a validator function and call it the start of a loader. DSL is pretty tricky with URLs, since they only have a single value type - `string`, so anything more complicated becomes very platform dependent. I.e. even something as simple as integer becomes murky because `1 === 1.00000000000000004`, but it's very unlikely `"1.00000000000000004"` should be a valid value where `"1"` is expected, only the person who writes the code for the route knows, not a framework.
